### ISSUE TITLE
Feature: Validation and Rollback on config:edit

### ISSUE THREAD

**Issue by aditya-manit** [iron-fish/ironfish#2618]
### Description

Current config:edit command has no validations, you can literally put anything into the config and save it.
It will just show

```
ironfish config:edit
Editing /Users/aditya.verma/.ironfish/config.json
```

When you do any other operations like config:show only then you know the config value that you had set wasn't right

```
 ironfish config:show
    ParseJsonError: Parsing config.json Failed
    Unexpected token "a" (0x61) in JSON at position 163 while parsing near "... \"databaseMigrate\": afalse\n}\n" in config.json 

      3 |     "editor": "/usr/bin/vim",
      4 |     "blockGraffiti": "9999Kingsuper",
    > 5 |     "databaseMigrate": afalse
        |                        ^
      6 | }
      7 |
```

To make the matter worse if you now try to edit config again it won't let you

```
ironfish config:edit
    ParseJsonError: Parsing config.json Failed
    Unexpected token "a" (0x61) in JSON at position 163 while parsing near "... \"databaseMigrate\": afalse\n}\n" in config.json 

      3 |     "editor": "/usr/bin/vim",
      4 |     "blockGraffiti": "9999Kingsuper",
    > 5 |     "databaseMigrate": afalse
        |                        ^
      6 | }
      7 |
```

Now you have to manually go into ~/.ironfish directory and edit config.json

## Summary
So to summarise:
1. config:edit gives no warning of invalid json or check any validation type errors
2. When you come to know about the error, it won't let you fix it

It's kinda of bug or miss, as other config operations have validation checks on them, I guess we are using an external editor to edit a file, things are a little trickier in this case and therefore it got missed.

**Comment by aditya-manit** [comment]
Solving this in https://github.com/iron-fish/ironfish/pull/2619
